Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py

```diff
@@ -1200,6 +1205,71 @@ def get_server_collections_command(integration_context):
     return result
 
 
+def create_relationships_objects(stix_iocs: list[dict[str, Any]]) -> list[dict[str, Any]]:
+    """
+    Create entries for the relationships returned by the searchRelationships command.
+    :param stix_iocs: Entries for the Stix objects associated with given indicators
+    :return: A list of dictionaries representing the relationships objects, including entityBs objects
+    """
+    relationships_list: list[dict[str, Any]] = []
+    iocs_value_to_id = {(stix_ioc.get('value') or stix_ioc.get('name')): stix_ioc.get('id') for stix_ioc in stix_iocs}
+    search_relationships = demisto.searchRelationships({'entities': list(iocs_value_to_id.keys())}).get('data', [])
+
+    demisto.debug(f"Found {len(search_relationships)} relationships for {len(iocs_value_to_id)} Stix IOC values.")
+    for relationship in search_relationships:
+
+        entity_b_value = relationship.get('entityB')
+        if entity_b_value not in iocs_value_to_id:
+            if entity_b_value and (entity_b_object := create_entity_b_stix_object(entity_b_value)):
```

Comment:
This is not good.
This will essentially run `searchIndicators` with `entity_b_value` for each unique relationship entity that is found.
Please see if you can batch all of the indicators you need to retrieve in one request - this should improve the performance.